File: devops-scm-spring-boot-starter/src/main/java/com/tencent/devops/scm/spring/properties/ScmProviderProperties.java
package com.tencent.devops.scm.spring.properties;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ScmProviderProperties {
    // 提供者编码
    private String providerCode;
    // 提供者类型
    private String providerType;
    // 是否需要代理
    private Boolean proxyEnabled;
    // 是否开启oauth2
    private Boolean oauth2Enabled;

    private HttpClientProperties httpClientProperties;
    // oauth2客户端属性
    private Oauth2ClientProperties oauth2ClientProperties;
    // webhook属性
    private WebhookProperties webhookProperties;
}

File: devops-scm-spring-boot-starter/src/main/kotlin/com/tencent/devops/scm/spring/properties/ScmProviderProperties.kt
package com.tencent.devops.scm.spring.properties

/**
 * 提供者属性
 */
data class ScmProviderProperties(
    // 提供者编码
    var providerCode: String? = null,
    // 提供者类型
    var providerType: String? = null,
    // 是否需要代理
    var proxyEnabled: Boolean? = null,
    // 是否开启oauth2
    var oauth2Enabled: Boolean? = null,
    var httpClientProperties: HttpClientProperties? = null,
    // oauth2客户端属性
    var oauth2ClientProperties: Oauth2ClientProperties? = null,
    // webhook属性
    var webhookProperties: WebhookProperties? = null
)